src/Microsoft.DotNet.Darc/src/Darc/Operations/GatherDropOperation.cs

```diff
@@ -337,6 +358,198 @@ public string GetFileShareLocationForReleaseJson(DownloadedBuild build)
+        ///     nupkgs\
+        ///         repo-shortname\
+        ///             packages\
+        ///                 shipping packages from repo
+        ///     sympkgs\
+        ///         repo-shortname\
+        ///             assets\
+        ///                 symbols\
+        ///                     symbol packages from repo
+        ///     nupkgs-all-just-for-reference.txt - file list of all packages, 
+        ///     nupkgs-aspnet.txt - file list of all packages that aspnetcore owns on nuget.org
+        ///     nupkgs-core.txt - file list of all packages that dotnet core owns on nuget.org
+        ///     sympkgs-all.txt - file list of all symbol packages
+        /// </remarks>
+        /// <returns>Async task</returns>
+        private void CreateReleasePackageLayout(List<DownloadedBuild> downloadedBuilds, string outputDirectory)
+        {
+            if (_options.DryRun || !_options.ReleaseLayout)
+            {
+                return;
+            }
+
+            // Identify the subdirectories. Do not need to create these now because
+            // we will create them as they go
+            string nupkgDirectory = Path.Combine(outputDirectory, nupkgDirectoryName);
+            string sympkgsDirectory = Path.Combine(outputDirectory, sympkgsDirectoryName);
+
+            // Universal text file content
+            StringBuilder allNupkgsFileContent = new StringBuilder();
+            StringBuilder sympkgsFileContent = new StringBuilder();
+
+            // Identity Csv contents
+            Dictionary<string, StringBuilder> identitiesFileContents = new Dictionary<string, StringBuilder>();
+
+            // core/wcf/aspnet nupkg lists
+            Dictionary<string, StringBuilder> nupkgFileContents = new Dictionary<string, StringBuilder>();
+
+            // Walk each input repo build and start filling out the directories
+            foreach (var downloadedBuild in downloadedBuilds)
+            {
+                // Start by determining whether we need to look at this build at all.
+                string repository = downloadedBuild.Build.GitHubRepository ?? downloadedBuild.Build.AzureDevOpsRepository;
+
+                if (!repositories.TryGetValue(repository, out (string, string) categoryAndShortName))
+                {
+                    continue;
+                }
+
+                string category = categoryAndShortName.Item1;
+                string shortName = categoryAndShortName.Item2;
+
+                // This goes into the release layout. Walk the downloaded assets
+                foreach (DownloadedAsset asset in downloadedBuild.DownloadedAssets)
+                {
+                    // If the asset is a shipping package, it goes into nupkgDirectory\<short name>\packages    
+                    if (!asset.Asset.NonShipping && asset.LocationType == LocationType.NugetFeed)
+                    {
+                        // Create the target directory
```

Comment:
The create target dir, file copy and line appending is quite similar in both branches of this if. May be the code could be put in a function.

Reply:
I had it that way originally and changed it back. It ends up the same-ish number of lines and seemed a bit contrived, since you still have to do additional thinks for the non-symbol packages.